pkg/services/searchV2/service_bench_test.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	user := &user.SignedInUser{
+		UserID: 1,
+		OrgID:  1,
+		Permissions: map[int64]map[string][]string{
+			1: {
+				dashboards.ActionDashboardsRead: dashScopes,
+				dashboards.ActionFoldersRead:    folderScopes,
+			},
+		},
+	}
+
+	return user
+}
+
+// Runs initial indexing of search service
+func runSearchService(searchService *StandardSearchService) error {
+	if err := searchService.dashboardIndex.buildInitialIndexes(context.Background(), []int64{int64(1)}); err != nil {
+		return err
+	}
+	searchService.dashboardIndex.initialIndexingComplete = true
+
+	// Required for sync that is called during dashboard search
+	go func() {
+		for {
+			doneCh := <-searchService.dashboardIndex.syncCh
+			close(doneCh)
+		}
+	}()
+
+	return nil
+}
+
+// Populates database with dashboards and folders
+func populateDB(folderCount, dashboardsPerFolder int, sqlStore *sqlstore.SQLStore) error {
+	// Insert folders
+	offset := 1
+	if errInsert := actest.ConcurrentBatch(actest.Concurrency, folderCount, actest.BatchSize, func(start, end int) error {
+		n := end - start
+		folders := make([]models.Dashboard, 0, n)
+		now := time.Now()
+
+		for u := start; u < end; u++ {
+			folderID := int64(u + offset)
+			folders = append(folders, models.Dashboard{
+				Id:       folderID,
+				Uid:      fmt.Sprintf("folder%v", folderID),
+				Title:    fmt.Sprintf("folder%v", folderID),
+				IsFolder: true,
+				OrgId:    1,
+				Created:  now,
+				Updated:  now,
+			})
+		}
+
+		err := sqlStore.WithDbSession(context.Background(), func(sess *db.Session) error {
+			if _, err := sess.Insert(folders); err != nil {
+				return err
+			}
+			return nil
+		})
+		return err
+	}); errInsert != nil {
+		return errInsert
+	}
+
+	// Insert dashboards
+	offset += folderCount
+	if errInsert := actest.ConcurrentBatch(actest.Concurrency, dashboardsPerFolder*folderCount, actest.BatchSize, func(start, end int) error {
+		n := end - start
+		dashboards := make([]models.Dashboard, 0, n)
+		now := time.Now()
+
+		for u := start; u < end; u++ {
+			dashID := int64(u + offset)
+			folderID := int64((u+offset)%folderCount + 1)
+			dashboards = append(dashboards, models.Dashboard{
+				Id:       dashID,
+				Uid:      fmt.Sprintf("dashboard%v", dashID),
+				Title:    fmt.Sprintf("dashboard%v", dashID),
+				IsFolder: false,
+				FolderId: folderID,
+				OrgId:    1,
+				Created:  now,
+				Updated:  now,
+			})
+		}
+
+		err := sqlStore.WithDbSession(context.Background(), func(sess *db.Session) error {
+			if _, err := sess.Insert(dashboards); err != nil {
+				return err
+			}
+			return nil
+		})
+		return err
+	}); errInsert != nil {
+		return errInsert
+	}
+
+	return nil
+}
+
+func benchSearchV2(b *testing.B, folderCount, dashboardsPerFolder int) {
+	searchService, testUser, err := setupBenchEnv(b, folderCount, dashboardsPerFolder)
+	require.NoError(b, err)
+
+	b.ResetTimer()
+
+	expectedResultCount := (dashboardsPerFolder + 1) * folderCount
```

Comment:
Nit. Apparently the results includes the general folder. It doesn't change much since the general folder was counted as the missing dashboard from the search.

```suggestion
	// Expect to get all folders + all dashboards + general folder
	expectedResultCount := (dashboardsPerFolder + 1) * folderCount + 1
```